Assets/Scripts/GameManager.cs

```diff
@@ -140,9 +160,85 @@ public void NextTrade()
 
     void EndMission()
     {
+        //check for failure conditions. if hull hp = 0, credits = 0, morale = 0
+        if (shipResources.hullHP <= 0)
+        {
+            string failedText = "OH NO! You ran out of hull. Have fun swimming home in space!";
+            missionFailure(failedText);
+        }
+        else if (shipResources.credits <= 0)
+        {
+            string failedText = "OH NO! You ran out of money. I hear McSpaceRonalds is hiring for 2 credits an hour!";
+            missionFailure(failedText);
+        }
+        else if (shipResources.crewMorale <= 0)
+        {
+            string failedText = "OH NO! Your crew has no morale and mutinied. Should've given them more time off or a second lunch break.";
+            missionFailure(failedText);
+        }
+        else
+        {
+            //what should this do?
+            //Make sure we can show the player they ended, what resources they got, and begin a new mission.
+            StartCoroutine(MissonEnds());
+        }
+
+    }
+
+    void missionFailure(string failedText)
+    {
+        MissionFailedPleaseContinue.gameObject.SetActive(false);
+        MissionFailedText.text = failedText;
+        MissionFailedScreen.SetActive(true);
+        StartCoroutine(MissionFailure());
+        /*create an ienumerator like the mission ends that resets the game
```

Comment:
I would remove old comments once development has made them no longer relevant. However you can also rework these as summaries of the work you completed at the top of methods.